docs/ibc/proposals.md

```diff
@@ -0,0 +1,39 @@
+<!--
+order: 5
+-->
+
+# Governance Proposals
+
+In uncommon situations, a highly valued client may become frozen due to uncontrollable 
+circumstances. A highly valued client, might have hundreds of channels being actively used.
+Some of those channels might have a significant amount of locked tokens used for ICS 20.
+
+If the one third of the validator set the chain the client represents decides to collude,
+they can sign off on two valid but conflicting headers each signed by the other one third
+of the honest validator sets. The light client can now be updated with two valid, but conflicting
+headers at the same height. The light client cannot know which header is trustworthy and therefore
+evidence of such misbehaviour is likely to be submitted resulting in a frozen light client. 
+
+Frozen light clients cannot be updated under any circumstance except via a governance proposal.
+Since validators can arbitarily agree to make state transitions that defy the written code, a 
+governance proposal has been added to ease the complexity of unfreezing or updating clients
+which have become "stuck". Unfreezing clients, re-enables all of the channels built upon that
+client. This may result in recovery of otherwise lost funds. 
+
+Tendermint light clients may also become expired if the trusting period has passed since their 
+last update. If a chain undergoes an unplanned upgrade, there may be no commitment to that upgrade
+signed by the validator set before the chain-id changes. In this situation, the validator set of
+the last valid update for the light client is never expected to produce another valid header since 
+the chain-id has changed, which will ultimately lead the on-chain light client to become expired.  
+
+In the case that a highly valued light client is frozen, expired, or rendered non-updateable, a
```

Comment:
In the cases where the client is frozen or had an unplanned upgrade, there may be many clients of the counterparty chain that are now non-updatable. The proposal should be able to specify multiple `subject-client-ids` as a string array. So that a single proposal can unfreeze all clients of the chain in question.

Reply:
I agree this seems practical, but I have some concerns:
- gas usage, is there a limit to gas used by a governance proposal? There could be a lot of consensus states and copying a lot of consensus states for multiple clients could be problematic in practice (especially if no one tests or simulates the results beforehand)
- the original ADR for this feature only specifies updating a single client, I believe because this should be used in rare cases

As I stated below, I think the trusting period needs to remain the same. It seems unlikely to me that there would exist identical clients that both need to be updated, since if they are identical, only 1 is needed

Reply:
- gas usage is an issue. I don't know when this would be executed. I guess at the end of a block? There's a general block gas limit. Though I don't know what happens to Endblock calculations if they go over it.
- I believe trusting period doesn't need to stay the same. But if we decide otherwise, then we can stick to a single subject client